Create CSS Entry for grid (#573)
Codecademy/docs: content/css/concepts/grids/terms/grid/grid.md

---
Title: 'grid'

> Description: 'Shorthand for setting grid property values.'

Reviewer: ```suggestion
Description: 'Shorthand for setting multiple grid property values.'
```

Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Grid'
  - 'Elements'
  - 'Properties'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

The `grid` property is shorthand for setting the following properties:

- [`grid-template-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-template-rows)
- [`grid-template-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-template-columns)
- `grid-template-areas`
- [`grid-auto-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-auto-rows)
- [`grid-auto-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-auto-columns)
- `grid-auto-flow`

## Syntax

```pseudo
grid: <grid-template-rows> / <grid-template-rows>;

grid: <grid-template-areas>;

grid: <grid-template-rows> / <grid-auto-columns>;

grid: <grid-auto-rows> / <grid-template-columns>;

grid: <grid-template-rows> / <grid-auto-flow> <grid-auto-columns>;

grid: <grid-auto-flow> <grid-auto-rows> / <grid-template-columns>;
```

## Example 1

> 

Reviewer 2: ```suggestion

Define a grid with three rows that are 200-pixel high, and two columns; the first column at `25%` wide, and the second at 75% wide.

```

> Define a grid with three rows that are 200-pixel high, and two columns; the first column at `25%` wide, and the second at 75% wide.

Reviewer: ```suggestion
In the example below is a `grid` declaration for the rows and column (separated by a `/`) with the following:

* There are three rows that are 200-pixels high each.
* There are two columns, with one being `25%` of the grid's width and another at `75%`-wide.
```

.my-grid {
  display: grid;
  grid: repeat(3, 200px) / 25% 75%;
}
```

> Defines a grid that has first three rows 200px tall, one column at 25% and the other column at 75%.
> 

Reviewer 2: ```suggestion
```

## Example 2

```css
.my-grid {
  display: grid;
  grid: auto / auto auto auto;
}
```

> Defines a grid that's three columns wide and has auto-sized rows and columns.

Reviewer: ```suggestion
Defines a grid that's three columns wide and has auto-sized rows and columns.
```
```suggestion
The example above defines a grid that has three columns. Each column and row is `auto`-sized.
```

---
Title: 'grid'
Description: 'Shorthand for setting multiple grid property values.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Grid'
  - 'Elements'
  - 'Properties'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

> The `grid` property is shorthand for setting the following properties:

Reviewer 2: ```suggestion
The `grid` property is a shorthand for setting the following properties:
```

- [`grid-template-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-template-rows)
- [`grid-template-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-template-columns)
- `grid-template-areas`
- [`grid-auto-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-auto-rows)
- [`grid-auto-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-auto-columns)
- `grid-auto-flow`

## Syntax

```pseudo
grid: <grid-template-rows> / <grid-template-columns>;

grid: <grid-template-areas>;

grid: <grid-template-rows> / <grid-auto-columns>;

grid: <grid-auto-rows> / <grid-template-columns>;

grid: <grid-template-rows> / <grid-auto-flow> <grid-auto-columns>;

grid: <grid-auto-flow> <grid-auto-rows> / <grid-template-columns>;
```

## Example 1

> In the example below is a `grid` declaration for the rows and column (separated by a `/`) with the following:

Reviewer 2: ```suggestion
The example below is a `grid` declaration for the rows and columns (separated by a `/`) with the following:
```

> - There are three rows that are 200-pixels high each.

Reviewer 2: ```suggestion
- There are three rows that are each 200-pixels high.
```
